src/braket/emulators/emulator_passes/criteria/emulator_criterion.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+
+from braket.emulators.emulator_passes.emulator_pass import EmulatorPass, ProgramType
+
+
+class EmulatorCriterion(EmulatorPass):
+    @abstractmethod
+    def validate(self, program: ProgramType) -> None:
+        """
+        An emulator criterion is used to perform some non-modifying validation
+        pass on an input program. Implementations of validate should return
+        nothing if the input program passes validation and raise an error otherwise.
+
+        Args:
+            program (ProgramType): The program to be evaluated against this criteria.
+        """
+        raise NotImplementedError
+
+    def run(self, program: ProgramType) -> ProgramType:
+        """
+        Validate the input program and return the program, unmodified.
+
+        Args:
+            program (ProgramType): The program to validate.
+
+        Returns:
+            ProgramType: The unmodified progam passed in as input.
+        """
+        self.validate(program)
+        return program
+
+    @abstractmethod
+    def __eq__(self, other: EmulatorCriterion) -> bool:
```

Comment:
Not making the `__eq__` method implementation required makes sense, it's definitely not necessary; it was originally there because I was interested in some notion of "emulator equality", but that wasn't really fleshed out. The method header/body has been removed from the ValidationPass class definition!